# Problema 1

Considere la sucesión $a_{n} = 3n^{2} + 7$. Por medio de un código en *Python* despliega en pantalla:

1. Los primeros $5$ términos.
2. Los $5$ términos que vienen inmediatamente después del decimoquinto término.

In [2]:
for i in range (1,16):
    s = 3*i**2 + 7
    print(f"Término {i}: {s}")

Término 1: 10
Término 2: 19
Término 3: 34
Término 4: 55
Término 5: 82
Término 6: 115
Término 7: 154
Término 8: 199
Término 9: 250
Término 10: 307
Término 11: 370
Término 12: 439
Término 13: 514
Término 14: 595
Término 15: 682


# Problema 2

Considere la sucesión $g_{n} = 5n^{3}$. Implemente un código en *Python* que muestre en pantalla:

1. Los primeros $4$ términos de la sucesión.
2. Los $4$ términos que vienen inmediatamente después del octavo término.
3. Determinar si el término $40.000$ pertenece a la sucesión. Si es así, indicar en qué posición se encuentra.

In [4]:
for i in range (1,100):
    s = 5*i**3
    print(f"Término {i}: {s}")
    if s == 40000:
        print(f"El término {i} es igual a 40000")
        break

Término 1: 5
Término 2: 40
Término 3: 135
Término 4: 320
Término 5: 625
Término 6: 1080
Término 7: 1715
Término 8: 2560
Término 9: 3645
Término 10: 5000
Término 11: 6655
Término 12: 8640
Término 13: 10985
Término 14: 13720
Término 15: 16875
Término 16: 20480
Término 17: 24565
Término 18: 29160
Término 19: 34295
Término 20: 40000
El término 20 es igual a 40000


# Problema 3

La sucesión de Fibonacci es una serie de números en la que cada número, desde el tercero en adelante, es la suma de los dos anteriores. Comienza con el $0$ y $1$, tal como se observa a continuación:

$$0, 1, 1, 2, 3, 5, 8, 13, ...$$

Formalmente, la sucesión se define de la siguiente manera:

\begin{split}
F(0) &= 0\\
F(1) &= 1\\
F(n) &= F(n - 1) + F(n - 2), \ \text{ para $n \geq 2$}
\end{split}

Esta sucesión se utiliza en una variedad de contextos en informática y algoritmos debido a sus propiedades matemáticas y patrones.

1. Implementa un código en *Python* que muestre en pantalla los primeros $20$ términos de la sucesión.
2. Calcula, utilizando un ciclo *for*, la suma de los primeros $30$ términos de la sucesión de Fibonacci.

In [5]:
f = [0, 1]
for i in range(2, 30):
    f.append(f[-1] + f[-2])

print("Primeros 20 términos de Fibonacci:", f[:20])

Primeros 20 términos de Fibonacci: [0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144, 233, 377, 610, 987, 1597, 2584, 4181]


# Problema 4

Implementa un código que permita guardar en una lista la sucesión de los primeros $n$ números pares, con $n \geq 1$. Luego, determina:

1. ¿Cuál es el valor del décimo término?
2. Calcula, utilizando un ciclo *for*, la suma de los primeros $100$ números pares.
3. ¿En qué posición se encuentra el número $58$ dentro de la sucesión?

In [19]:
pares = []
for i in range (1,201):
    if i % 2 == 0:
        pares.append(i)

for i in pares:
    if pares[i] == 58:
        print(f"El número 58 se encuentra en la posición {i+1} de la lista de números pares")
        break

print("El decimo valor de la los primeros 100 números pares es:", pares[9])
print(f"La suma de los primeros 100 numeros pares es {sum(pares)}")


El número 58 se encuentra en la posición 29 de la lista de números pares
El decimo valor de la los primeros 100 números pares es: 20
La suma de los primeros 100 numeros pares es 10100


# Problema 5

Un nuevo software tiene $500$ usuarios y se espera que el número de usuarios crezca un $15\%$ respecto al mes anterior. Considere que $b_{n}$ es el número de usuarios transcurridos $n$ meses.

1. Calcula cuántos usuarios tendrá el software al cabo de tres meses. ¿Y al cabo de seis meses?
2. Mediante un código en *Pyhton* muestra la cantidad de usuarios que tendrá el software cada mes durante el primer año.
2. ¿Cuántos usuarios tendrá en total el software luego de un año? Utilice ciclos *for* para calcular.

In [21]:
suma = 0
for i in range (1,13):
    s = 500*(1.15)**i
    suma += s
    print(f"Mes {i}: {s}")
print(f"La suma de los 12 meses es: {suma}")

Mes 1: 575.0
Mes 2: 661.2499999999999
Mes 3: 760.4374999999999
Mes 4: 874.5031249999997
Mes 5: 1005.6785937499997
Mes 6: 1156.5303828124995
Mes 7: 1330.0099402343744
Mes 8: 1529.5114312695305
Mes 9: 1758.9381459599597
Mes 10: 2022.7788678539534
Mes 11: 2326.195698032046
Mes 12: 2675.125052736853
La suma de los 12 meses es: 16675.958737649216


# Problema 6

Un estudiante de ingeniería informática planea ahorrar para un proyecto personal. Para esto, el primer mes deposita en una cuenta $\$12.000$, el segundo mes deposita $\$14.000$ y así sucesivamente, de modo que cada mes aumenta el depósito en $\$2.000$. Considere que $a_{i}$ corresponde al depósito del estudiante en el mes $i$.

1. Implementa un código que permita mostrar en pantalla lo que depositó mensualmente durante el primer año.
2. ¿Cuánto depositó el estudiante en febrero del segundo año?
3. ¿Cuál será el total ahorrado al finalizar dos años?

In [32]:
suma = 0
for i in range (1,25):
    s = 12000 + 2000*(i-1)
    suma += s
    print(f"Mes {i}: {s}")
print(f"La suma de los 24 meses es: {suma}")

Mes 1: 12000
Mes 2: 14000
Mes 3: 16000
Mes 4: 18000
Mes 5: 20000
Mes 6: 22000
Mes 7: 24000
Mes 8: 26000
Mes 9: 28000
Mes 10: 30000
Mes 11: 32000
Mes 12: 34000
Mes 13: 36000
Mes 14: 38000
Mes 15: 40000
Mes 16: 42000
Mes 17: 44000
Mes 18: 46000
Mes 19: 48000
Mes 20: 50000
Mes 21: 52000
Mes 22: 54000
Mes 23: 56000
Mes 24: 58000
La suma de los 24 meses es: 840000


# Problema 7

Por lo general, los proyectos son organizados en varias etapas que deben completarse secuencialmente.

La fase inicial de un proyecto de software tiene una duración de $3$ meses y cada fase subsecuente reduce su duración en un $10 \%$ debido a la experiencia ganada. Determine:

1. ¿Cuánto dura la quinta fase?
2. Calcula la duración total del proyecto después de $10$ fases.

In [39]:
for i in range (1,11):
    s = 3 * (1-0.1)**(i-1)
    print(f"Fase {i}: {s} meses")

Fase 1: 3.0 meses
Fase 2: 2.7 meses
Fase 3: 2.43 meses
Fase 4: 2.1870000000000003 meses
Fase 5: 1.9683000000000002 meses
Fase 6: 1.7714700000000003 meses
Fase 7: 1.5943230000000002 meses
Fase 8: 1.4348907000000002 meses
Fase 9: 1.2914016300000002 meses
Fase 10: 1.1622614670000002 meses


# Problema 8

En una sucesión aritmética, la diferencia entre un término y el anterior es $4$ y el décimo término es $-20$. Determine:

1. El primer término de la sucesión.
2. La expresión del término de lugar $n$.
3. Utilice un código en *Python* para calcular el término que ocupa el lugar $100$.
4. Determinar la posición del número $1.680$ en la secuencia. Calcule utilizando un ciclo *while*.

In [44]:
a10 = -20  
d = 4 
n = 10 

a1 = a10 - (n - 1) * d
print(a1) 

for i in range (1,1001):
    s = -56 + 4*(i-1)
    print(f"Término {i}: {s}")
    if s == 1680:
        print(f"El término {i} es igual a 1680")
        break

-56
Término 1: -56
Término 2: -52
Término 3: -48
Término 4: -44
Término 5: -40
Término 6: -36
Término 7: -32
Término 8: -28
Término 9: -24
Término 10: -20
Término 11: -16
Término 12: -12
Término 13: -8
Término 14: -4
Término 15: 0
Término 16: 4
Término 17: 8
Término 18: 12
Término 19: 16
Término 20: 20
Término 21: 24
Término 22: 28
Término 23: 32
Término 24: 36
Término 25: 40
Término 26: 44
Término 27: 48
Término 28: 52
Término 29: 56
Término 30: 60
Término 31: 64
Término 32: 68
Término 33: 72
Término 34: 76
Término 35: 80
Término 36: 84
Término 37: 88
Término 38: 92
Término 39: 96
Término 40: 100
Término 41: 104
Término 42: 108
Término 43: 112
Término 44: 116
Término 45: 120
Término 46: 124
Término 47: 128
Término 48: 132
Término 49: 136
Término 50: 140
Término 51: 144
Término 52: 148
Término 53: 152
Término 54: 156
Término 55: 160
Término 56: 164
Término 57: 168
Término 58: 172
Término 59: 176
Término 60: 180
Término 61: 184
Término 62: 188
Término 63: 192
Término 64: 196
Término 65:

# Problema 9

El tercer término de una sucesión geométrica es $5$ y el sexto término es $40$. Determine:

1. La razón geométrica de la sucesión.
2. El primer término de la sucesión.
3. La expresión del término de lugar $n$.
4. Utilice un código en *Python* para calcular el término que ocupa el lugar $8$.
4. ¿Qué lugar ocupa en esta sucesión el término $20.480$? Calcule utilizando un ciclo *while*.

In [45]:
a3 = 5
a6 = 40

r = (a6 / a3) ** (1/3)
print(f"La razón geométrica es: {r}")

a1 = a3 / r**2
print(f"El primer término es: {a1}")

def term(n, a1, r):
    return a1 * r**(n-1)

n = 8
term_8 = term(n, a1, r)
print(f"El término en la posición {n} es: {term_8}")


a_n = 20480
n = 1

while term(n, a1, r) != a_n:
    n += 1

print(f"El término 20,480 ocupa la posición: {n}")


La razón geométrica es: 2.0
El primer término es: 1.25
El término en la posición 8 es: 160.0
El término 20,480 ocupa la posición: 15


# Problema 10

Una empresa de tecnología está optimizando sus servidores. Para ello mide las transacciones por segundo (TPS) de sus servidores en cada mes. Se ha obtenido que en el segundo mes, el rendimiento del servidor fue $1,2$ miles de TPS y en el quinto mes, el rendimiento fue $2,7$ miles de TPS. Si el número de TPS que hay mes a mes sigue una sucesión geométrica, determina utilizando un código en *Python*:

1. ¿Cuál es el rendimiento el noveno mes?
2. ¿En qué mes el rendimiento es de $23,4$ miles de TPS?

Utilice dos decimales para el cálculo.

# Problema 11

Una aplicación de redes sociales se estrenó el $1$ de enero con una cierta cantidad de usuarios. A partir de ahí, el número de usuarios nuevos de cada día fue siempre el mismo. El $3$ de enero tenía $500$ usuarios en total y el $10$ de enero tenía $1.501$ usuarios totales.Determine:

1. ¿Con cuántos usuarios comenzó a funcionar la aplicación?
2. ¿Cuántos usuarios tendrá la aplicación el $31$ de enero?
3. Si la aplicación tiene $5.505$ usuarios, ¿cuántos meses lleva funcionando?



# Problema 12

Una empresa de publicidad ofrece para un puesto de trabajo un salario inicial y un aumento mensual en una razón constante. Javier se presenta a la entrevista, y el encargado del departamento de recursos humanos le explica que en el segundo mes de trabajo su sueldo será de $\$402.000$, y que en el tercer mes de trabajo será de $\$404.010$. Determine:

1. El salario aproximado que Javier recibiría en el décimo mes de trabajo en la empresa.
2. ¿Cuál sería el ingreso total recibido por los sueldos de Javier durante el tercer año de trabajo?

# Problema 13

Una empresa de ciberseguridad capacita trabajadores cada trimestre. En el primer trimestre capacitaron a $30$ trabajadores, en el segundo trimestre a $60$ trabajadores, y en el tercer trimestre a $120$ trabajadores. Si en los siguientes trimestres la empresa mantiene el mismo patrón de crecimiento, determine utilizando *Python*:

1. ¿A cuántos trabajadores capacitarán en el octavo trimestre?
2. Si la empresa lleva en el mercado $3$ años, ¿a cuántos trabajadores han capacitado en total?

# Problema 14

Al llenar un tambor con agua, se registra que durante el minuto $4$ el tambor recibe $128$ litros, y durante el minuto $9$ recibe $88$ litros. Sabiendo que en cada minuto recibe una cantidad $d$ de litros de agua menos que en el minuto anterior, determine:

1. ¿En qué instante el tambor deja de recibir agua?
2. ¿Cuántos litros recibe en total el tambor?

Realice los cálculos anteriores utilizando *Python*.